civis/io/_tables.py

```diff
@@ -694,21 +689,26 @@ def dataframe_to_civis(df, database, table, api_key=None, client=None,
         warnings.warn("`archive` is deprecated and will be removed in v2.0.0. "
                       "Use `hidden` instead.", FutureWarning)
 
+    delimiter = '|'
+
     with TemporaryDirectory() as tmp_dir:
         tmp_path = os.path.join(tmp_dir, 'dataframe_to_civis.csv')
-        to_csv_kwargs = {'encoding': 'utf-8', 'index': False}
+        to_csv_kwargs = {
+            'encoding': 'utf-8', 'index': False, 'header': True,
+            'sep': delimiter, 'quoting': csv.QUOTE_ALL, 'escapechar': '\\',
```

Comment:
Is there a reason we need `csv.QUOTE_ALL` instead of the default `csv.QUOTE_MINIMAL`? This adds more bytes to files and increases there size. I would opt for the default unless there is some use case we are aware of.

Quick note on `escapechar`, this is not used unless `quoting=csv.QUOTE_NONE` and `doublequote=False`, so someone would have to explicitly change the defaults on both of those for this to take effect. The default value is `None` which seems appropriate here.

Reply:
Quote all is needed because of the back slash quote issue breaking platform import. Everything needs to be wrapped in quotes to avoid having as few backslashes in the file as possible.

The escapechar thing I can change.

Reply:
There's an issue with [platform import](https://civisanalytics.atlassian.net/browse/CIVP-12730) where if there's a backslash and a doublequote next to each other anywhere in the file for any reason, the import will fail. The JIRA has my code demonstrating the problem and the fix I've been using for the MobileCommons & Classy intergrations.

I'm doing a PR for these settings because they're the settings that work with importing any kind of extremely messy text data (aside from the two edge cases in the JIRA) into platform. Right now, it's extremely easy to have imports that fail with no easy way to determine the issue.

Reply:
@ffaristocrat I ran through this example in detail for the last 2 hours and updated linked JIRA. I still do not agree that quote all is needed or helping resolve the issues presented in the example on the JIRA. If anything, for your example you should set `doublequote=False`.